### ISSUE TITLE
Bug: An interface can only extend an object type or intersection of object types with statically known members.ts(2312) on type export 

### ISSUE THREAD

**Issue by Nut41tank** [eslint/eslint#16744]
### Environment

Node version: v18.12.1
npm version: v8.19.2
Local ESLint version: v8.31.0
Global ESLint version: Not found
Operating System: win32 10.0.1904

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>
today lint tell that incorrect extends type but SelectProps is already export as type alias 

```js
@mui
export type SelectProps<T = unknown> = CommonProps<T> & ConditionalRenderValueType<T>;

import Select, { SelectChangeEvent, SelectProps } from '@mui/material/Select';
interface DropdownProps<T> extends SelectProps {
  ...
  handleChange: (event: SelectChangeEvent<unknown | string>) => void;
}
```

### What did you expect to happen?

should not face 
An interface can only extend an object type or intersection of object types with statically known members.ts(2312)

### What actually happened?

face this error 
An interface can only extend an object type or intersection of object types with statically known members.ts(2312)

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by kecrily** [comment]
`@typescript-eslint/parser` is not maintained by the ESLint team, [go to typescript-eslint feedback](https://github.com/typescript-eslint/typescript-eslint)
